#11038: Changed runner-labels

ttmchiou opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/11038

### Problem description
Too many labels on runners. clean up labels and improve transparency

### What's changed
Replaced "grayskull" and "wormhole_b0" with "in-service" 
Replaced "multi-chip*" and "multi-pci*" with product codes E150/N150/N300 
Removed "runner-test"
Removed "arch-wormhole_b0" "arch-grayskull" from single-chip cards
Removed "perf-wormhole_b0" and replaced with generic "pipeline-perf"
Added generic pipeline-stress label and removed "stress-wormhole_b0" "stress-grayskull"
Added cloud-virtual-machine labels to pipelines for VMs (since bare-metal machines also carry some product codes)
to prevent 

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

ttmchiou committed:
#11038: Changed runner-labels
Replaced "grayskull" and "wormhole_b0" with "in-service"
Removed "multi-chip*" and "multi-pci*" with product codes E150/N150/N300
Removed "runner-test"
Removed "arch-wormhole_b0" "arch-grayskull"

TT-billteng reviewed (approved):
!!!!

ttmchiou committed:
#11038: Added missing cloud-virtual-machine and in-service tags

ttmchiou committed:
#11038: added pipeline-stress label and removed stress-{} label

ttmchiou commented:
@tt-rkim Is removing the "arch-grayskull" and "arch-wormhole_b0" going to be an issue?

tt-rkim commented:
@ttmchiou 
For single-card no, but for T3000 and Galaxy, recall that we need to know what arch they are. They won't always be wormhole.

We should keep arch for those.

ttmchiou committed:
#11038: removed t3k label on t3k perplexity (old artifact)

TT-billteng commented on .github/workflows/build-and-unit-tests.yaml:
```diff
@@ -39,7 +39,9 @@ jobs:
-      - "in-service"
+      - cloud-virtual-machine
+      - in-service
+      - config-single-card
```
isn't this label redundant?

ttmchiou replied:
Wondering if it makes sense to have this config-single-card labelled paired with card (ie E150/N300,etc)
Since Raymond mentioned that we should keep arch for t3000 and galaxy since they won't always be wormhole cards, does it make sense to instead use card or product labels?

Examples are because T3Ks also have N300s and Galaxies have N150s 

we also have t3k VMs coming up so putting cloud-virtual-machine and config-t3000 with N300 label could conflict.

imo, it would be more flexible keeping configuration labels for what a machine should have 
VM/BM
Card installed (e150/n150/n300/etc)
configuration (single/t3k/tg/tgg)

Unless, these are set product configurations with set cards, in which case, config-single-card can be removed.

Thoughts?
@TT-billteng @tt-rkim

ttmchiou replied:
on second thought,
let's table that discussion for another issue, if we want to reorganize labels.
I'll remove it for now since we can use combinations of
VM/BM + card

tt-rkim replied:
I like ensuring all runners have the 4:

- VM/BM
- card installed
- config (single/t3k/tg/tgg)
- pipeline purpose

However, one wrinkle is that Galaxy doesn't necessarily have a specific card type installed within it. We could use the intermediary driver card as a proxy (ie. current setup uses N150 in the middle). However, this may not always be the case. We should check syseng and @ubcheema to make sure we know what's going on.

Also don't break data pipelines. Check `infra/data_collection/github/utils.py:176` to about 42 lines down.

TT-billteng replied:
feel like we're overcomplicating this at this point in time

VM/BM
config: E150/N150/N300/T3k/tgg/tgg
pipeline purpose (for special cases)

ttmchiou committed:
#11038: Replaced self-reset label with bare-metal since identical

ttmchiou committed:
#11038: Added arch detection using single_cards to not break data pushup

Want to avoid breaking "[internal] Produce data for external analysis"
by removing arch-{}, we will raise an exception in arch not being pushed
up

ttmchiou committed:
#11038: removed arch-{} labels from single-chip runners

ttmchiou committed:
#11038: Added arch tags to t3k/tg/tgg workflows & cleaned up t3k workflows
